src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +282,93 @@ def test_set_beam_shift(self):
         self.scanner.beamShift.value = (new_beam_shift_x, new_beam_shift_y)
         self.assertAlmostEqual((new_beam_shift_x, new_beam_shift_y), self.scanner.beamShift.value)
 
+    def test_apply_autostigmator(self):
+        """
+        Test for the auto functionality of the apply_autostigmator.
+        Runs the apply_autostigmator multiple times and checks (using "get_stigmator()) if the found stigmator values
+        are equal each run. Checks the returned state, the return object and if an error is raised for an incorrect input state.
+        """
+        # Create sets holding stigmator values to check if the outcome is consistent
+        unique_stigmator_values_x = set()  # Set holding only the unique stigmator values in x direction
+        unique_stigmator_values_y = set()  # Set holding only the unique stigmator values in y direction
+
+        # Run apply_autostigmator multiple times and afterwards check if the outcome is the same for each run
+        for i in range(0, 5):
+            autostigmator_return = self.scanner.apply_autostigmator("electron1", "run")
+            time.sleep(0.2)  # Give microscope/simulator the time to update the state
+            # direct_call = self.microscope.set_autostigmator("electron1", "run")
+            autostigmator_state = self.microscope.is_autostigmating("electron1")
+            self.assertEqual(autostigmator_state, True)
+            self.assertIsInstance(autostigmator_return, ProgressiveFuture)
+
+            autostigmator_return = self.scanner.apply_autostigmator("electron1", "stop")
+            time.sleep(1.0)  # Give microscope/simulator the time to update the state
+            autostigmator_state = self.microscope.is_autostigmating("electron1")
+            self.assertEqual(autostigmator_state, False)
+            self.assertIsInstance(autostigmator_return, ProgressiveFuture)
+
+            unique_stigmator_values_x.add(self.microscope.get_stigmator()[0])
+            unique_stigmator_values_y.add(self.microscope.get_stigmator()[1])
+
+        # Check if all the stigmator values are the same for each run
+        self.assertEqual(len(unique_stigmator_values_x), 1,
+                         "'apply_autostigamtor' does not provide the same stigmator values each time in x direction.")
+        self.assertEqual(len(unique_stigmator_values_y), 1,
+                         "'apply_autostigamtor' does not provide the same stigmator values each time in y direction.")
+
```

Comment:
No they need to be up to 14 decimals the same.
Do you want to include some extra margin? An if yes how much?
With the current check if these values are exactly the same it works.

I found that set check up to 14 decimals by using:
foo = set()
i=30
while len(foo)<2:
     foo.add(1 + 10**-i)
     i = i - 1# Counting down from 30 decimals up to the point an extra element is added to the set.
print("Precision of set is %s" % i)

Reply:
I wonder if exactly the same value is always found for autostigmation, if you would test on the hardware. But I think this is fine for now.

Reply:
There is no way that you're going to run 5 times the auto stigmator on a real hardware and get twice the same value. Let alone 5 times the same value! I guess you get always the same value because you're using a simulator. 
In general, I don't think we need to test that the stigmator behaviour is good anyway. This is up to TFS, if their auto stigmator is good, then it's nice, if their stigmator doesn't always work, it's a pity.... but we can't do much about it, so we don't need to care  in the test case.
So, here, I'd suggest to just check that the values are roughly the same (eg, 20% -> `almost_equal(...,rtol=0.2)`)